server/src/main/java/org/elasticsearch/cluster/metadata/ProjectMetadata.java

```diff
@@ -2109,7 +2080,7 @@ public static ProjectMetadata fromXContent(XContentParser parser) throws IOExcep
                     switch (currentFieldName) {
                         case "reserved_state" -> {
                             while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
-                                projectBuilder.put(ReservedStateMetadata.fromXContent(parser));
+                                ReservedStateMetadata.fromXContent(parser);
                             }
```

Comment:
This applies to full cluster restart or restoring a snapshot. Can we add a TODO (or `@FixForMultiProject` annotation) here to remove this leniency once existing MP clusters all upgraded to a point where this is not necessary? For wire communication, we get the reminder from the tranport version.